src/piadm.sh

```diff
@@ -486,20 +487,48 @@ update_boot_sectors() {
 
 	some=0
 	for a in "${boot_devices[@]}"; do
-		# Plow through devices, even if some fail.  installboot also
-		# does loader-into-EFI-System-Partition this way.  Trailing /
-		# is important in the -b argument because boot is actually a
-		# symlink.
-		installboot -m -b /${bootfs}/boot/ /${bootfs}/boot/pmbr \
-			/${bootfs}/boot/gptzfsboot \
-			/dev/rdsk/${a}${suffix} > /dev/null 2>&1 || \
-			eecho "WARNING: Can't installboot on ${a}${suffix}"
+		if [[ "$flag" == "-d" ]]; then
+			if [[ "$suffix" == "s0" ]]; then
+				# BIOS boot, we don't care.
+				continue
+			fi
+			# otherwise mount the ESP and trash it.
+			tdir=`mktemp -d`
```

Comment:
One small nit -- aren't we preferring $() syntax over backtics?

Reply:
Doh!  Good catch.